scapy/plist.py

```diff
@@ -75,7 +78,7 @@ def __getattr__(self, attr):
         return getattr(self.res, attr)
     def __getitem__(self, item):
         if isinstance(item,type) and issubclass(item,BasePacket):
-            return self.__class__(filter(lambda x: item in self._elt2pkt(x),self.res),
+            return self.__class__(list(filter(lambda x: item in self._elt2pkt(x),self.res)),
```

Comment:
Using filter, it is needed, but we could use list comprehension to remove list and filter